src/devTools.js

```diff
@@ -112,13 +118,18 @@ export function connect(options = {}) {
     },
     send: (action, payload) => {
       if (action) {
-        send(action, payload, options, 'ACTION', id);
+        send(action, payload, options, 'ACTION', id, hostnamePromise);
       } else {
-        send(undefined, payload, options, 'STATE', id);
+        send(undefined, payload, options, 'STATE', id, hostnamePromise);
       }
     },
     error: (payload) => {
-      socket.emit({ type: 'ERROR', payload, id: socket.id, instanceId: id });
+      // like in the `send` function
+      setTimeout(async () => {
+        // makes sure the connection is established
+        await start(options, hostnamePromise);
```

Comment:
If the promise inside of the `start` function fails, the error is just being rethrown here(no need to handle), is this ok?

Reply:
First, see my response above where you asked the same question.

Secondly, I don't understand why you introduced `setTimeout` here. If you feel strongly that `error` cannot be an async function (I don't understand why that is, but hopefully you can explain), then you should just use an async "IIFE":

```js
(async () => {
  // makes sure the connection is established
  await start(options, hostnamePromise);
  socket.emit({ type: 'ERROR', payload, id: socket.id, instanceId: id });
})();
```